### PR TITLE
[core] Use enum instead of string for file.format

### PR DESCRIPTION
**Pull request by FangYongs**
### Purpose

Use enum instead of string for file.format and validate the format value in ddl

### Tests

1. Added the test `SchemaManagerTest.testInvalidFormatType` to validate the format value

### API and Format 

*(Does this change affect API or storage format)* no

### Documentation

*(Does this change introduce a new feature)* no

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/CoreOptions.java [hunk 1]
```diff
@@ -956,6 +962,31 @@ public InlineElement getDescription() {
         }
     }
 
+    /** Specifies the file format type for store. */
+    public enum FileFormatType implements DescribedEnum {
+        ORC("orc", "ORC format for file store."),
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/CoreOptions.java:967]: maybe remove `file store`? now log store is only in flink

File: paimon-core/src/main/java/org/apache/paimon/schema/SchemaValidation.java [hunk 2]
```diff
@@ -100,6 +101,9 @@ public static void validateTableSchema(TableSchema schema) {
             }
         }
 
+        // Valid the format type
+        checkNotNull(options.formatType());
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/schema/SchemaValidation.java:105]: option with default value should never return null.
we don't need this checking.

**Comment by FangYongs** [comment]
Thanks @JingsongLi  Updated

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**